### PR TITLE
UI: Use icons from theme on Linux

### PR DESCRIPTION
**Pull request by cg2121**
Fixes:
https://obsproject.com/mantis/view.php?id=937

### CODE REVIEW

**Comment by DDRBoxman** [comment]
Should this be future proofed so themes can define the active tray icon?

**Comment by cg2121** [comment]
@DDRBoxman I have now implemented it now to where the a OBS active tray icon can be used if available.

File: UI/obs-app.cpp [hunk 1]
```diff
@@ -1040,11 +1040,26 @@ bool OBSApp::InitTheme()
 	return SetTheme("Default");
 }
 
+QIcon OBSApp::SetIcon(bool tray)
```
Comment (DDRBoxman) [inline UI/obs-app.cpp]: This bool should probably be called "active" or something of the sort

File: UI/window-projector.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ OBSProjector::OBSProjector(QWidget *widget, obs_source_t *source_, int monitor,
 	type           = type_;
 
 	if (isWindow) {
-		setWindowIcon(QIcon(":/res/images/obs.png"));
+		setWindowIcon(App()->SetIcon());
```
Comment (DDRBoxman) [inline UI/window-projector.cpp]: Should this use the tray icon or the app icon the theme provides?

File: UI/obs-app.cpp [hunk 3]
```diff
@@ -1040,11 +1040,26 @@ bool OBSApp::InitTheme()
...
+{
+	QIcon icon;
+
+	if (!tray)
+		icon = QIcon::fromTheme("obs", QIcon(":/res/images/obs.png"));
```
Comment (DDRBoxman) [inline UI/obs-app.cpp]: Probably should be obs-tray.

**Comment by cg2121** [comment]
The three different icons should be accounted for now. I have also greatly simplified the code.

File: UI/window-basic-main.cpp [hunk 4]
```diff
@@ -6639,8 +6641,8 @@ void OBSBasic::ToggleShowHide()
 
 void OBSBasic::SystemTrayInit()
 {
-	trayIcon.reset(new QSystemTrayIcon(QIcon(":/res/images/obs.png"),
-			this));
+	trayIcon.reset(new QSystemTrayIcon(QIcon::fromTheme("obs-tray",
+					QIcon(":/res/images/obs.png")), this));
```
Comment (RytoEX) [inline UI/window-basic-main.cpp]: Style:  Indent with two tabs for a line continuation.

```suggestion
			QIcon(":/res/images/obs.png")), this));
```

**Comment by varlesh** [comment]
Thx @cg2121 :heart: 
Worked fine:
image
image

**Pull request merged**